eureka/src/main/java/com/linecorp/armeria/client/eureka/EurekaEndpointGroup.java

```diff
@@ -388,6 +391,23 @@ private static Endpoint endpoint(InstanceInfo instanceInfo, boolean secureVip) {
         if (ipAddr != null && hostname != ipAddr) {
             endpoint = endpoint.withIpAddr(ipAddr);
         }
+        endpoint = endpoint.withMetadata(metadata(instanceInfo));
         return endpoint;
     }
+
+    private static AttributeMap metadata(InstanceInfo instanceInfo) {
+        final AttributeMap metadata = new DefaultAttributeMap();
+
+        if (instanceInfo.getHealthCheckUrl() != null) {
+            metadata.attr(AttributeKey.valueOf("healthCheckUrl")).set(instanceInfo.getHealthCheckUrl());
```

Comment:
- Just out of curiosity, why do you need `healthCheckUrl`?
- Is it better to add `InstanceInfo` itself to metedata? I feel that it is hard to standardize the attributes for Eureka.

Reply:
We may use   'healthCheckUrl ' to create HealthcheckEndpointGroup. It seems adding InstanceInfo itself to metadata is better.

Reply:
Do we need an option to disable adding eureka metadata to attrs?

Reply:
> Do we need an option to disable adding eureka metadata to attrs?

Sounds good to me. 👍

Reply:
InstanceInfo is in the internal package,  We should move them to a public package later (not this MR)?
I think the data model is stable.